### ISSUE TITLE
placeholder problem in tensorflow

### ISSUE THREAD

**Issue by fritss** [bethgelab/foolbox#298]
[foolbox.zip](https://github.com/bethgelab/foolbox/files/3098309/foolbox.zip)

I have trained a model with tensorflow.keras and saved it to a .h5 file.
To make adversarial examples with foolbox I run:

image_placeholder = tf.placeholder(tf.float32, shape=(None, 32, 32, 3))
logits = cifarmodel.output
model = foolbox.models.TensorFlowModel(image_placeholder, logits, (0, 1))
label = np.argmax(model.predictions(image))

(the whole python program and the modelfile are attached.)

This results in the following error message:
InvalidArgumentError (see above for traceback): You must feed a value for placeholder tensor 'conv2d_1_input' with dtype float and shape [?,32,32,3]

It looks like the new placeholder is not attached properly to the model input.
What do I wrong?

**Comment by jonasrauber** [comment]
From the code above it seems that you just create a placeholder but your model doesn't actually use it (and possibly has it's own placeholder that never receives data and thus throws an error). Maybe looking at the TensorFlow examples helps: https://foolbox.readthedocs.io/en/latest/user/examples.html#tensorflow-vgg19
Please reopen if this is still an issue.

**Comment by FernandoZhuang** [comment]
## Problem

At first, i followed this [tutorial](https://foolbox.readthedocs.io/en/v2.3.0/user/examples.html#tensorflow-vgg19) (foolbox 2.3.0 TensorFlow CIFAR10)
```python
image_placeholder= tf.placeholder(tf.float32, shape=(None, 32, 32, 3),)

fmodel = fb.models.TensorFlowModel(image_placeholder, logits, bounds=bounds, )

fmodel.forward_one(img)
```

Same complain!

## Solution

we should use the same input placeholder from our model itself

```python
inputs = [placeholder for op in tf.get_default_graph().get_operations() if op.type == 'Placeholder' for placeholder in op.values()]
image_placeholder = inputs[0]

fmodel = fb.models.TensorFlowModel(image_placeholder, logits, bounds=bounds,)

fmodel.forward_one(img)
```
## Reference

1. [How to use keras model.predict with placeholders input?](https://stackoverflow.com/questions/58645384/how-to-use-keras-model-predict-with-placeholders-input)
2. [In Tensorflow, get the names of all the Tensors in a graph](https://stackoverflow.com/questions/36883949/in-tensorflow-get-the-names-of-all-the-tensors-in-a-graph)
